crates/bevy_render/src/camera/visible_entities.rs

```diff
@@ -258,3 +350,61 @@ pub fn visible_entities_system(
         // to prevent holding unneeded memory
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use bevy_ecs::{
+        schedule::{Schedule, Stage, SystemStage},
+        system::IntoSystem,
+        world::World,
+    };
+    use bevy_transform::hierarchy::{parent_update_system, BuildWorldChildren};
+
+    #[test]
+    fn propagates_visibility() {
```

Comment:
This test should add at least one more level of hierarchy to ensure effective visibility propagates all the way down the tree.

Reply:
Improved the test as requested.